### PR TITLE
[SC-111] Deploy to multiple environments

### PR DESCRIPTION
**Pull request by zaro0508**
Setup travis to deploy the synpase login application to multiple
Beanstalk environments.

Merges to develop branch deploys to scipooldev while merges
to prod branch deploys to scipoolprod.

### CODE REVIEW

File: .elasticbeanstalk/config.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 branch-defaults:
-  master:
+  develop:
```
Comment (brucehoff) [inline .elasticbeanstalk/config.yml:2]: Will we keep the 'master' branch up to date, or just ignore it?

File: .travis.yml [hunk 2]
```diff
@@ -16,14 +13,20 @@ install:
   - pip install --upgrade pre-commit awsebcli
 stages:
   - name: test
-  - name: deploy
-    if: type = push
+  - name: deploy-scipooldev
+    if: type = push AND branch = develop
+  - name: deploy-scipoolprod
+    if: type = push AND branch = prod
```
Comment (brucehoff) [inline .travis.yml:19]: In this context, is a PR merge considered the same as a "push"?  Who will have the authority to push/merge into prod?  Just you, @zaro0508 ?

File: .travis.yml [hunk 3]
```diff
@@ -16,14 +13,20 @@ install:
...
   include:
     - stage: test
       script:
         - mvn clean test
-    - stage: deploy
+    - stage: deploy-scipooldev
+      script:
+        - mvn clean package
+        - eb deploy synapse-login-scipooldev --profile default --region us-east-1 --verbose
+    - stage: deploy-scipoolprod
```
Comment (brucehoff) [inline .travis.yml:29]: Is the first parameter in 'eb deploy' (i.e. synapse-login-scipooldev) the name of an AWS account?

File: README.md [hunk 4]
```diff
@@ -69,6 +69,20 @@ We have configured Travis CI to automatically build, test and deploy the applica
 ## Contributions
 Contributions are welcome
 
+## Development Workflow
+This project has two branches `develop`, and `prod`.  When proposing changes to this project
+the workflow is to propose the changes to develop branch then promote to prod branch.
```
Comment (brucehoff) [inline README.md]: Why do you say, "propose the changes to" instead of "create a pull request against"?

File: README.md [hunk 5]
```diff
@@ -69,6 +69,20 @@ We have configured Travis CI to automatically build, test and deploy the applica
...
+
+step by step:
+1. Propose PR to develop branch
+2. Review and approve the PR then merge it to develop branch
+3. CI/CD system builds, test, and deploys artifact from develop branch to AWS dev
+environment in org-sagebase-scipooldev
```
Comment (brucehoff) [inline README.md]: "in org-sagebase-scipooldev" -> "in the in org-sagebase-scipooldev AWS account"

**Comment by brucehoff** [comment]
Will this repo' (synapse-login-scipool) subsume the repo' synapse-login-scipoolprod?  If so, I like it because it seems better that a repo' should NOT refer only to dev or prod (that's the role of the multiple branches in a repo').

**Comment by zaro0508** [comment]
Yes, that is correct.  That's why i had to rename the repo, but instead of rename i created a new repo and will remove synapse-login-scipoolprod after SC-111 is complete

File: .elasticbeanstalk/config.yml (hunk 1 above)
Comment (zaro0508) [reply .elasticbeanstalk/config.yml:2]: it's already been nuked.

File: .travis.yml (hunk 3 above)
Comment (zaro0508) [reply .travis.yml:29]: no, it's the name of the Elastic Beanstalk environment.  usually it's like dev, stage, prod, etc.. but in this case i called it `synapse-login-scipooldev`

File: README.md (hunk 4 above)
Comment (zaro0508) [reply README.md]: isn't it the same?  if you think `create` is more clean then i can update with that.  doesn't matter to. me.

File: .travis.yml (hunk 2 above)
Comment (zaro0508) [reply .travis.yml:19]: yes, merge to a branch is a `push`.  users/groups who have write access to the repo have the ability to push/merge to branches.

File: .travis.yml (hunk 3 above)
Comment (brucehoff) [reply .travis.yml:29]: I see.  How then does Travis-CI know which AWS account to push to?

File: .travis.yml (hunk 2 above)
Comment (zaro0508) [reply .travis.yml:19]: it's done with setup_aws_cli.sh in line #15. That file determines which credentials to use (scipooldev or scipoolprd) based on the which the branch that's being built.

File: .travis.yml (hunk 3 above)
Comment (zaro0508) [reply .travis.yml:29]: it's done with setup_aws_cli.sh in line #15. That file determines which credentials to use (scipooldev or scipoolprd) based on the which the branch that's being built.

**Review by brucehoff — APPROVED** [review]
Looks good.  Thank you for answering all my questions.

**Pull request merged**